src/mapbox_logo.js

```diff
@@ -22,6 +22,11 @@ var MapboxLogoControl = L.Control.extend({
         if (json.mapbox_logo) {
             L.DomUtil.addClass(this._container, 'mapbox-logo-true');
         }
+
+        // account for usage of styleJSON as functional tileJSON for defaults
+        if (!json.tilejson && (json.owner && json.owner === 'mapbox')) {
```

Comment:
I think the `json.owner` check is superfluous; you'd be okay with

`if (!json.tilejson && json.owner === 'mapbox') {`

since we expect `json` to be an object and if there's no `owner` key, it'll just return `undefined` for `json.owner`.

Reply:
Good point. 👍 simplified in 97179796